src/TACSAssembler_thread.cpp

```diff
@@ -101,11 +102,27 @@ void *TACSAssembler::assembleRes_thread(void *t) {
         aux_count++;
       }
 
-      // Add the residual from the auxiliary elements
-      while (aux_count < naux && aux[aux_count].num == elemIndex) {
-        aux[aux_count].elem->addResidual(elemIndex, assembler->time, elemXpts,
-                                         vars, dvars, ddvars, elemRes);
-        aux_count++;
+      // Add the residual from any auxiliary elements, if the load factor is 1
+      // they can be added straight to the elemRes, otherwise they need to be
+      // scaled first
+      if (lambda == 1.0) {
```

Comment:
This should be: if (TacsRealPart(lambda) == 1.0) {
Won't work in complex mode

Reply:
Alternatively, we could make lambda a `double`. Is there a reason we'd need to complex step over it?

Reply:
This is true, I did think you could complex step the load factor to extract the aux element part of the residual, but there are probably better ways to do that. I will switch it back to `double` to be consistent with the `alpha`, `beta`, `gamma` scaling factors

Reply:
nvm, I was wrong, `alpha`, `beta` and `gamma` are in fact TacsScalars, so I'll keep lambda as a TacsScalar too